tests/shared/process-event.ts

```diff
@@ -1273,6 +1273,80 @@ export const createProcessEventTests = (
         expect(person2.is_identified).toBe(true)
     })
 
+    test('we do not alias users if distinct id changes but we are already identified', async () => {
```

Comment:
It's also worth testing manually with stateful client libraries whether this behavior works.

E.g. Given:

event1
Identify to userA
event2
Identify to userB
event3

What will the result be, which events will userA and userB own?

Reply:
Will add some events into the test and assert they are associated with the right Persons

Reply:
On the plugin server level the tests you added are :ok_hand: - we should still 100% check and verify the client libraries behavior as well though. 

This is a huge behavioral change and we want to be sure we understand it fully :)

Reply:
Yup, I had a play with trying to get the plugin-server and posthog-js running locally but wasn't convinced it was on configured correctly.

One of the other libraries would be ideal as well as posthog-js already has the guard against the issue behaviour. I guess rudderstack would be a good candidate here?

Reply:
Also, are there any integration tests I should add to also?

Reply:
Other libraries are stateless, though I haven't checked out e.g. react native/ios libraries.

Perhaps the thing to do is temporarily remove the guard from posthog-js and try this out without it?

Re integration test suite: https://github.com/PostHog/posthog-js/blob/master/testcafe/e2e.spec.js does kind of this already, though it might be a bit hard to modify.

Reply:
yup I'll have a go with disabling the guard and see if I can get it running against local.